### ISSUE TITLE
ReactRouterSSR.run + ReactHelmet; No SEO during server render

### ISSUE THREAD

**Issue by HugoCe17** [thereactivestack/kickstart-meteor-react-router#84]
### TL;DR ###
I am working on a website that needs SEO to be top notch and currently, Meta information isn't being rendered on the server and ReactHelmet doesn't seem to be working while following Documentation, perhaps there is something I am not seeing. Any help would be much appreciated!

### `curl localhost:3000` yields ###
```html
<!DOCTYPE html>
<html>
<head>  <link rel="stylesheet" type="text/css" class="__meteor-css__" href="/7a0c3da108620d2fa80b2dff7094a91a213f90bb.css?meteor_css_resource=true">
</head>
<body><div id="react-app"><div data-reactroot="" data-reactid="1" data-react-checksum="1362567650"><!-- react-empty: 2 --><p data-reactid="3"> Hello, World </p></div></div>



<script type="text/inject-data">%7B%22fast-render-data%22%3A%7B%22collectionData%22%3A%7B%7D%2C%22subscriptions%22%3A%7B%7D%7D%7D</script>
<script type="text/javascript">__meteor_runtime_config__ = JSON.parse(decodeURIComponent("%7B%22meteorRelease%22%3A%22METEOR%401.3.2.4%22%2C%22meteorEnv%22%3A%7B%22NODE_ENV%22%3A%22production%22%2C%22TEST_METADATA%22%3A%22%7B%7D%22%7D%2C%22PUBLIC_SETTINGS%22%3A%7B%7D%2C%22ROOT_URL%22%3A%22http%3A%2F%2Flocalhost%3A3000%2F%22%2C%22ROOT_URL_PATH_PREFIX%22%3A%22%22%2C%22appId%22%3A%22a247lv2k0c6tl3kml5%22%2C%22autoupdateVersion%22%3A%225e51d3616b9ea247247dd266d6f21c4f56e4ba3a%22%2C%22autoupdateVersionRefreshable%22%3A%225fec34bc5b4cf41bbf0b2aa9c139366d0584e3d3%22%2C%22autoupdateVersionCordova%22%3A%22none%22%7D"));</script>

  <script type="text/javascript" src="/b6699ee43222b2ae75c1be136b7cdc119da26b21.js?meteor_js_resource=true"></script>


</body>
</html>
```

I set the routes so the Helmet would be the root to the App and index route to be what users first see.

### src/EcomApp/client ##
```javascript
import { Route, IndexRoute } from 'react-router';
import EcomApp from './components/EcomApp/EcomApp';
import EcomHelmet from './components/EcomHelmet/EcomHelmet';

export default (
  <Route path="/" component={ EcomHelmet }>
    <IndexRoute component={ EcomApp }/>
  </Route>
);

```

### EcomHelmet.js ##
```javascript
import React from 'react'
import Helmet from 'react-helmet'

export default function EcomHelmet({children}) {
  return (
    <div>
      <Helmet
        title="ReactiveCommerce"
        defaultTitle="ReactiveCommerce"
        meta={[
          { name: "keywords", content: 'Ecommerce Store Front with Shopify SDK' },
          { name: "msapplication-TileColor", content: "#ffffff" },
          { name: "msapplication-TileImage", content: "/ms-icon-144x144.png" },
          { name: "theme-color", content: "#ffffff" },
          { name: "apple-mobile-web-app-capable", content: "yes" },
          { name: 'description', content: 'An Ecommerce storefront project leveraging meteor, react, Shopify SDK to deliver a great Material Designed Ecommerce experience' },
          { name: "apple-mobile-web-app-status-bar-style", content: "black-translucent" },
          { name: 'viewport', content: 'width=device-width, initial-scale=1, minimum-scale=1, maximum-scale=1, user-scalable=0' }
        ]}
        link={[
          { rel: "apple-touch-icon", sizes: "57x57", href: "./io-favi/apple-icon-57x57.png" },
          { rel: "apple-touch-icon", sizes: "60x60", href: "./io-favi/apple-icon-60x60.png" },
          { rel: "apple-touch-icon", sizes: "72x72", href: "./io-favi/apple-icon-72x72.png" },
          { rel: "apple-touch-icon", sizes: "76x76", href: "./io-favi/apple-icon-76x76.png" },
          { rel: "apple-touch-icon", sizes: "114x114", href: "./io-favi/apple-icon-114x114.png" },
          { rel: "apple-touch-icon", sizes: "120x120", href: "./io-favi/apple-icon-120x120.png" },
          { rel: "apple-touch-icon", sizes: "144x144", href: "./io-favi/apple-icon-144x144.png" },
          { rel: "apple-touch-icon", sizes: "152x152", href: "./io-favi/apple-icon-152x152.png" },
          { rel: "apple-touch-icon", sizes: "180x180", href: "./io-favi/apple-icon-180x180.png" },
          { rel: "icon", type: "image/png", sizes: "192x192", href: "./io-favi/android-icon-192x192.png" },
          { rel: "icon", type: "image/png", sizes: "32x32", href: "./io-favi/favicon-32x32.png" },
          { rel: "icon", type: "image/png", sizes: "96x96", href: "./io-favi/favicon-96x96.png" },
          { rel: "icon", type: "image/png", sizes: "16x16", href: "./io-favi/favicon-16x16.png" },
          { rel: "manifest", href: "./io-favi/manifest.json" },
          { href: "https://fonts.googleapis.com/icon?family=Material+Icons", rel: "stylesheet" },
          { rel: "stylesheet", href: "https://maxcdn.bootstrapcdn.com/font-awesome/4.5.0/css/font-awesome.min.css" },
          { href: 'https://fonts.googleapis.com/css?family=Varela', rel: 'stylesheet', type: 'text/css' }
        ]}/>
      {children}
    </div>
  )
}
```

This is where I get lost. According to the `ReactRouterSSR.run` docs, `htmlHook(html): newHtml` is to be called as part of the `serverOption` object yet the complex example illustrates `htmlHook(html): newHtml` being called inside the `prop` object pertaining to `clientOptions` object. 

Here is my current implementation. 

### src/routes.js ##

```javascript
import { Route } from 'react-router';
import { ReactRouterSSR }  from 'meteor/reactrouter:react-router-ssr';
import ReactHelmet from 'react-helmet';
import EcomHelmet from 'EcomApp/client/components/EcomHelmet/EcomHelmet'
import ecomRoutes from 'EcomApp/client/routes';
import ReactCookie from 'react-cookie';


ReactRouterSSR.Run(ecomRoutes, {
  props: {
    onUpdate() {
      // Notify the page has been changed to Google Analytics
      ga('send', 'pageview');
    },
    htmlHook(html) {
      const head = ReactHelmet.rewind();
      return data.replace('<head>', '<head>' + head.title + head.base + head.meta + head.link + head.script);
    }
  }
}, {
    preRender: function (req, res) {
      ReactCookie.plugToRequest(req, res);
      global.navigator = {
        userAgent: req.headers['user-agent']
      };
    }
  }
);


if (Meteor.isClient) {
  // Load Google Analytics
  (function (i, s, o, g, r, a, m) {
    i['GoogleAnalyticsObject'] = r; i[r] = i[r] || function () {
      (i[r].q = i[r].q || []).push(arguments)
    }, i[r].l = 1 * new Date(); a = s.createElement(o),
      m = s.getElementsByTagName(o)[0]; a.async = 1; a.src = g; m.parentNode.insertBefore(a, m)
  })(window, document, 'script', 'https://www.google-analytics.com/analytics.js', 'ga');

  ga('create', 'UA-77710690-1', 'auto');
  ga('send', 'pageview');
}
```

**Comment by ribbedcrown** [comment]
Hi, @HugoCe17, it seems, examples are outdated... But this works for me:
```javascript

import { Route } from 'react-router';
import { ReactRouterSSR }  from 'meteor/reactrouter:react-router-ssr';
import Helmet from 'react-helmet';
import AppRoutes from 'App/routes';

ReactRouterSSR.Run(
  // routes
  <Route>
    {AppRoutes}
  </Route>,
  // clientOptions
  {
    // ...
  },
  // serverOptions
  {
    htmlHook(html) {
      const head = Helmet.rewind();
      return html.replace('<head>', '<head>' + head.title + head.base + head.meta + head.link + head.script);
    }
  }
);
```

**Comment by HugoCe17** [comment]
That worked @ribbedcrown
